resources/ext/ezbake.conf

```diff
@@ -2,11 +2,36 @@ ezbake: {
   pe: {}
   foss: {
     redhat: { dependencies: ["puppet-agent >= 5.99.0"],
-              preinst:  ["if rpm -q puppetdb | grep ^puppetdb-2.* > /dev/null && [ $1 -eq 2 ] ; then tar -czf /tmp/puppetdb-upgrade-config-files.tgz -C /etc/puppetdb/conf.d config.ini database.ini jetty.ini ; fi" ],
+              preinst:  [
+                  "if rpm -q puppetdb | grep '^puppetdb-2.*' > /dev/null && [ \"$1\" -eq 2 ]; then",
+                  "    tar -czf /tmp/puppetdb-upgrade-config-files.tgz -C /etc/puppetdb/conf.d config.ini database.ini jetty.ini",
+                  "fi",
+                  "# puppetdb changed its main namespace in 6.3 and the current rpm",
+                  "# upgrade strategy fails to restart the server correctly because",
+                  "# it ends up using the templated stop from the new package which ",
+                  "# refers to the wrong namespace for the older pdb that's still running.",
+                  "(maybe_dead=''",
+                  " pid=\"$(pgrep -f \"puppetdb.* -m puppetlabs.puppetdb.main\")\"",
+                  " if test $? -eq 0; then",
+                  "     kill \"$pid\" 2>/dev/null",
+                  "     for i in {1..75}; do  # Wait up to ~15s",
+                  "         if kill -0 \"$pid\" 2>/dev/null; then",
+                  "             maybe_dead=1",
+                  "             break",
+                  "         fi",
+                  "         sleep 0.2",
+                  "     done",
+                  "     if test -z \"$maybe_dead\"; then",
+                  "         echo 'Unable to kill puppetdb server cleanly; sending KILL' 1>&2",
+                  "         kill -9 \"$pid\"",
+                  "     fi",
```

Comment:
A `sleep 1` right here, before the `service start`, appears to avoid some kind of race that prevents the start from working.